scripts/survey/pushcompletedsubmissions.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * name : pushcompletedsubmissions.js
+ * author : Saish Borkar
+ * created-date : 18-JULY-2024
+ * Description : Migration script to push completed survey submissions to kafka
+ */
+
+const path = require("path");
+let rootPath = path.join(__dirname, '../../')
+require('dotenv').config({ path: rootPath+'/.env' })
+require("../../config/globalVariable")();
+
+let _ = require("lodash");
+let mongoUrl = process.env.MONGODB_URL;
+
+let dbName = mongoUrl.split("/").pop();
+let url = mongoUrl.split(dbName)[0];
+var MongoClient = require('mongodb').MongoClient;
+var ObjectId = require('mongodb').ObjectID;
+const args = process.argv.slice(2);
+const surveySubmissionsHelper = require(MODULES_BASE_PATH + "/surveySubmissions/helper");
+const fs = require('fs');
+const utils = require('../../generics/helpers/utils')
+let IDString = args[0];
+
+let IDArray = IDString.split(',');
+
+let successArray = [];
+(async () => {
+
+    let connection = await MongoClient.connect(url, { useNewUrlParser: true, useUnifiedTopology: true });
+    let db = connection.db(dbName);
+
+    try {
+        // if it is a solution id
+      let firstRecordSet = await db
+        .collection("surveySubmissions")
+        .find({solutionId:{$in:IDArray.map(id=>ObjectId(id))},status:"completed"})
+        .toArray(); 
+
+      let secondRecordSet = await db
+        .collection("surveySubmissions")
+        .find({_id:{$in:IDArray.map(id=>ObjectId(id))},status:"completed"})
+        .toArray(); 
+
+    if(firstRecordSet && firstRecordSet.length > 0)
```

Comment:
@borkarsaish65 If they pass both solutionId and submissionId conditions won't be executed. Please check how they will pass the ids. Based on discussion if changes required please add it here.

Reply:
Checked with Prashant , only submission_ids will be pushed.